src/app/components/SelectNode/UploadMacaroons.tsx

```diff
@@ -1,171 +1,216 @@
+                />
+              </Form.Item>
+              <Form.Item label="Readonly macaroon">
+                <Input
+                  name="readonly"
+                  value={readonly}
+                  onChange={this.handleChangeMacaroonHex}
+                  placeholder="Paste hex string here"
+                />
+              </Form.Item>
+            </>
+          ) : (
+            <>
+              <Upload.Dragger
+                accept=".macaroon"
+                showUploadList={false}
+                beforeUpload={
+                  (file) => this.handleMacaroonUpload('admin', file)
+                }
+              >
+                <p className="ant-upload-drag-icon">
+                  <Icon type={admin ? 'check-circle' : 'inbox'}/>
+                </p>
+                <p className="ant-upload-text">
+                  Upload <code>admin.macaroon</code>
+                </p>
+                <p className="ant-upload-hint">
+                  Click or drag to upload macaroon
+                </p>
+              </Upload.Dragger>
+
+              <Upload.Dragger
+                accept=".macaroon"
+                showUploadList={false}
+                beforeUpload={
+                  (file) => this.handleMacaroonUpload('readonly', file)
+                }
+              >
+                <p className="ant-upload-drag-icon">
+                  <Icon type={readonly ? 'check-circle' : 'inbox'}/>
+                </p>
+                <p className="ant-upload-text">
+                  Upload <code>readonly.macaroon</code>
+                </p>
+                <p className="ant-upload-hint">
+                  Click or drag to upload
+                </p>
+              </Upload.Dragger>
+
+              {dirs &&
               <div className="UploadMacaroons-hint">
-                Macaroons are usually located in the following places
-                <br/>
-                <strong>macOS</strong>: <code>{dirs.MACOS}</code>
-                <br/>
-                <strong>Windows</strong>: <code>{dirs.WINDOWS}</code>
-                <br/>
-                <strong>Linux</strong>: <code>{dirs.LINUX}</code>
+                  Macaroons are usually located in the following places
+                  <br/>
+                  <strong>macOS</strong>: <code>{dirs.MACOS}</code>
+                  <br/>
+                  <strong>Windows</strong>: <code>{dirs.WINDOWS}</code>
+                  <br/>
+                  <strong>Linux</strong>: <code>{dirs.LINUX}</code>
               </div>
-            }
-          </>
-        )}
+              }
+            </>
+          )}
 
-        {error &&
+          {error || checkAuthError &&
           <Alert
-            message="Invalid macaroon"
-            description={error}
-            type="error"
-            closable
-            showIcon
+              message="Invalid macaroon"
+              description={error || checkAuthError}
```

Comment:
It looks like this is causing it to try rendering an `Error` object which causes the following error in React:
```
Uncaught Invariant Violation: Objects are not valid as a React child (found: Error: Read-only macaroon did not authenticate). If you meant to render a collection of children, use an array instead.
```
The screen blanks out when this happens. I think you want the `.message` attribute instead.